File: frontend/src/types/project.ts
export interface Project {
  id: string
  slug: string
  name: string
  description?: string
  color: string
  is_active: boolean
  member_count: number
  created_at: string
  updated_at: string
}

export interface ProjectCreate {
  slug: string
  name: string
  description?: string
  color?: string
}

export interface ProjectUpdate {
  name?: string
  description?: string
  color?: string
  is_active?: boolean
}

export interface ProjectMember {
  id: string
  project_id: string
  admin_id: string
  role: 'master_admin' | 'admin' | 'member'
  created_at: string
  admin_email?: string
  admin_name?: string
}

export interface ProjectMemberWithAdmin {
  id: string
  project_id: string
  admin_id: string
  role: string
  created_at: string
  admin_email: string
  admin_name: string
}

export interface ProjectStats {
  project_id: string
  member_count: number
  master_admin_count: number
  admin_count: number
  member_role_count: number
}

export interface MyProjectsResponse {
  projects: Project[]
  is_super: boolean
}

export const PROJECT_ROLES = ['master_admin', 'admin', 'member'] as const
export type ProjectRole = typeof PROJECT_ROLES[number]

File: backend/app/schemas/project.py
import uuid
from datetime import datetime
from pydantic import BaseModel, Field


class ProjectCreate(BaseModel):
    slug: str = Field(min_length=2, max_length=50, pattern=r"^[a-z0-9-]+$")
    name: str = Field(min_length=1, max_length=100)
    description: str | None = None
    color: str = "#6366f1"


class ProjectUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    color: str | None = None
    is_active: bool | None = None


class ProjectResponse(BaseModel):
    id: uuid.UUID
    slug: str
    name: str
    description: str | None = None
    color: str
    is_active: bool
    member_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ProjectListResponse(BaseModel):
    items: list[ProjectResponse]
    total: int


class ProjectMemberCreate(BaseModel):
    admin_id: uuid.UUID
    role: str = "member"  # master_admin | admin | member


class ProjectMemberUpdate(BaseModel):
    role: str  # master_admin | admin | member


class ProjectMemberResponse(BaseModel):
    id: uuid.UUID
    project_id: uuid.UUID
    admin_id: uuid.UUID
    role: str
    created_at: datetime

    model_config = {"from_attributes": True}


class ProjectMemberWithAdmin(BaseModel):
    id: uuid.UUID
    project_id: uuid.UUID
    admin_id: uuid.UUID
    role: str
    created_at: datetime
    admin_email: str
    admin_name: str

    model_config = {"from_attributes": True}


class ProjectStats(BaseModel):
    project_id: uuid.UUID
    member_count: int
    master_admin_count: int
    admin_count: int
    member_role_count: int


class MyProjectsResponse(BaseModel):
    projects: list[ProjectResponse]
    is_super: bool